group-generators/helpers/data-providers/farcaster/index.ts

```diff
@@ -0,0 +1,95 @@
+import axios from "axios";
+import { retryRequest } from "@group-generators/helpers/data-providers/utils/utils";
+import { FetchedData } from "topics/group";
+
+export class FarcasterProvider {
+  url: string;
+  headers: {
+    authorization: string;
+    accept: string;
+  };
+
+  public constructor() {
+    this.url = "https://api.farcaster.xyz/v2/";
+    this.headers = {
+      authorization: process.env.FARCASTER_API_KEY as string,
+      accept: "application/json",
+    };
+  }
+
+  public async getLastCreatedFid(): Promise<number> {
+    const { data: res } = await axios({
+      url: this.url + "recent-users",
+      method: "get",
+      headers: this.headers,
+    });
+
+    if (Object(res).result) {
+      return Object(res).result.users[0].fid;
+    } else {
+      throw new Error(Object(res));
+    }
+  }
+
+  public async resolveAddress(context: any, fid: number): Promise<string> {
+    // skip the 0 fid because it doesn't exist
+    if (fid == 0) {
+      return "";
+    }
+    let res: any;
+    try {
+      const { data: res } = await axios({
+        url: context.url + "verifications?fid=" + fid,
```

Comment:
you could extract this in an other function because there is always the same pattern 
```typescript
await axios({
      url: this.url + "recent-users",
      method: "get",
      headers: this.headers,
    });
    ```